### ISSUE TITLE
Got error "Invalid use of BasicClientConnManager: connection still allocated."

### ISSUE THREAD

**Issue by gavrilfb** [dzieciou/curl-logger#57]
Using curl-logger version 2.0.0 with restassured version 4.3.1 and apache http client 4.5.3 (the same thing also happened using the latest version of both)
when I tried to hit api in parallel, it throws this error.

> java.lang.IllegalStateException: java.lang.IllegalStateException: Invalid use of BasicClientConnManager: connection still allocated.
Make sure to release the connection before allocating another one.

you can use below code to replicate the error

> RestAssured.config = RestAssuredConfig.config().
                sslConfig(SSLConfig.sslConfig().relaxedHTTPSValidation()).
                httpClient(HttpClientConfig.httpClientConfig().
                        setParam("http.connection.timeout", 10000).
                        setParam("http.socket.timeout", 10000).
                        setParam("http.connection-manager.timeout", 10000)).
                encoderConfig(encoderConfig().appendDefaultContentCharsetToContentTypeIfUndefined(false));
        RestAssured.config = CurlRestAssuredConfigFactory.updateConfig(RestAssured.config); // removing this part will throws no error
        List<String> test = new ArrayList<>();
        test.add("1");
        test.add("2");
        test.add("3");
        test.add("4");
        test.add("5");
        test.parallelStream().forEach(i -> given().get("https://run.mocky.io/v3/2c47d181-5d7c-4fb8-81d5-e06265c91203"));

**Comment by dzieciou** [comment]
Sounds like a duplicate of #37.

**Comment by gavrilfb** [comment]
Okay, seems to be working correctly on 2.0.1
Thank you
